File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(int a, int b) {
    string str_a = to_string(a);
    string str_b = to_string(b);
    return str_a + str_b > str_b + str_a;
}

string solution(vector<int> numbers) {
    string answer = "";
    sort(numbers.begin(), numbers.end(), cmp);
    
    for (int& i : numbers)
        answer += to_string(i);
    
    if (answer[0] == '0')
        return "0";
    
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def cmp(a, b):
    if a + b > b + a:
        return -1
    elif a + b < b + a:
        return 1
    else:
        return 0

def solution(numbers):
    answer = ''
    strnums = list(map(str, numbers))
    sortedstrnums = sorted(strnums, key = cmp_to_key(cmp))
    
    for t in sortedstrnums:
        answer += t
    
    return '0' if answer[0] == '0' else answer